### PR TITLE
Adds equals/hashCode/toString overrides to Tag

### PR DESCRIPTION
**Pull request by dinooliva**

### CODE REVIEW

File: core/javatests/com/google/instrumentation/stats/TagTest.java [hunk 1]
```diff
@@ -31,6 +31,29 @@ public void testTagTest() {
     assertThat(tag.getValue()).isEqualTo(V1);
   }
 
+  @Test
+  public void testEquals() {
+    assertThat(Tag.create(K1, V1)).isEqualTo(Tag.create(K1, V1));
+    assertThat(Tag.create(K1, V1)).isNotEqualTo(Tag.create(K2, V1));
+    assertThat(Tag.create(K1, V1)).isNotEqualTo(Tag.create(K1, V2));
```
Comment (sebright) [inline core/javatests/com/google/instrumentation/stats/TagTest.java]: This test could use `EqualsTester`.

Comment (dinooliva) [inline core/javatests/com/google/instrumentation/stats/TagTest.java]: I thought about that but I'm mostly checking isNotEqualsTo() so I don't think it adds anything.

Comment (sebright) [inline core/javatests/com/google/instrumentation/stats/TagTest.java]: EqualsTester can also test that Objects unequal.  We use it for a similar test in TagKeyTest: https://github.com/google/instrumentation-java/blob/2e621e42394633c0a4a75cb59d54e176f66b646e/core/javatests/com/google/instrumentation/stats/TagKeyTest.java#L49

File: core/java/com/google/instrumentation/stats/Tag.java [hunk 2]
```diff
@@ -38,6 +38,26 @@ public TagValue getValue() {
     return value;
   }
 
+  @Override
+  public boolean equals(Object obj) {
```
Comment (bogdandrutu) [inline core/java/com/google/instrumentation/stats/Tag.java]: Guava has a nice utility for equals/hashCode tests com.google.common.testing.EqualsTester;

Comment (dinooliva) [inline core/java/com/google/instrumentation/stats/Tag.java]: Yes, I've already switched over to that.

File: core/javatests/com/google/instrumentation/stats/TagTest.java [hunk 3]
```diff
@@ -31,6 +32,35 @@ public void testTagTest() {
...
+        .testEquals();
+  }
+
+  @Test
+  public void testToString() {
+    new EqualsTester()
+        .addEqualityGroup(Tag.create(K1, V1).toString(), "Tag<k1,v1>")
+        .addEqualityGroup(Tag.create(K2, V1).toString(), "Tag<k2,v1>")
+        .addEqualityGroup(Tag.create(K1, V2).toString(), "Tag<k1,v2>")
+        .testEquals();
```
Comment (sebright) [inline core/javatests/com/google/instrumentation/stats/TagTest.java]: I just meant `testEquals()`.  I think `testToString()` was clearer before.

Comment (dinooliva) [inline core/javatests/com/google/instrumentation/stats/TagTest.java]: Done.

**Pull request merged**